src/main/java/org/springframework/data/elasticsearch/repository/support/AbstractElasticsearchRepository.java

```diff
@@ -167,44 +175,41 @@ public long count() {
 	}
 
 	@Override
-	public <S extends T> Iterable<S> save(Iterable<S> entities) {
+	public <S extends T> Iterable<S> saveAll(Iterable<S> entities) {
 		Assert.notNull(entities, "Cannot insert 'null' as a List.");
-		if (!(entities instanceof Collection<?>)) {
-			throw new InvalidDataAccessApiUsageException("Entities have to be inside a collection");
-		}
-		List<IndexQuery> queries = new ArrayList<IndexQuery>();
+		List<IndexQuery> queries = new ArrayList<>();
 		for (S s : entities) {
 			queries.add(createIndexQuery(s));
 		}
 		elasticsearchOperations.bulkIndex(queries);
-		elasticsearchOperations.refresh(entityInformation.getIndexName(), true);
+		elasticsearchOperations.refresh(entityInformation.getIndexName());
 		return entities;
 	}
 
 	@Override
-	public boolean exists(ID id) {
-		return findOne(id) != null;
+	public boolean existsById(ID id) {
+		return findById(id) != null;
```

Comment:
`findById` returns an `Optional` and not a null value
also see https://jira.spring.io/browse/DATAES-363